WordPress/Classes/ViewRelated/Post/PostEditorState.swift

```diff
@@ -186,41 +176,68 @@ public class PostEditorStateContext {
         self.delegate = delegate
 
         guard let originalPostStatus = originalPostStatus else {
-            editorState = PostEditorStatePublish()
+            action = .publish
             return
         }
 
         switch originalPostStatus {
         case .draft where userCanPublish == false:
-            editorState = PostEditorStateSubmitForReview()
+            action = .submitForReview
         default:
-            editorState = PostEditorStateUpdate()
+            action = .update
+
         }
     }
 
     /// Call when the post status has changed due to a remote operation
     ///
     func updated(postStatus: BasePost.Status) {
         currentPostStatus = postStatus
-        let updatedState = editorState.updated(postStatus: postStatus, context: self)
-        guard type(of: editorState) != type(of: updatedState) else {
-            return
-        }
 
-        editorState = updatedState
+        let updatedAction = { () -> PostEditorAction in
```

Comment:
> If you agree that's the case, I can compare values inside action's didSet and make a decision there instead (so that you can set it safely without extra checks.

Yeah, that sounds like a reasonable solution! 👍